src/api/groups.js

```diff
@@ -0,0 +1,67 @@
+import express from 'express';
+
+import { configureUsersService } from '../config.js';
+import { groupSchema, validateBody } from './validation.js';
+
+const groupsRouter = express.Router();
+const usersService = configureUsersService();
+
+groupsRouter.route('/')
+    .post(validateBody(groupSchema), async (req, res, next) => {
+        const group = req.body;
+
+        try {
+            const groupData = await usersService.addGroup(group);
+            res.json(groupData);
+        } catch (error) {
+            return next(error);
+        }
+    })
+    .get(async (req, res) => {
+        const groups = await usersService.getAllGroups();
```

Comment:
any exception crashes your endpoint if there is not try ... catch block.
Please, check every route and add try catch
<img width="806" alt="image" src="https://user-images.githubusercontent.com/48351153/217201388-3a69a7cd-8449-451f-8a77-98202b2a5617.png">

It happens because express allows use async functions but it can't handle exceptions throws from it. So you need to use try catch everywhere

Reply:
wrapped all endpoints in try/catch